store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/HBaseStore.java

```diff
@@ -0,0 +1,198 @@
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.id.ElementId;
+import uk.gov.gchq.gaffer.data.element.id.EntityId;
+import uk.gov.gchq.gaffer.hbasestore.operation.handler.AddElementsHandler;
+import uk.gov.gchq.gaffer.hbasestore.operation.handler.GetAdjacentIdsHandler;
+import uk.gov.gchq.gaffer.hbasestore.operation.handler.GetAllElementsHandler;
+import uk.gov.gchq.gaffer.hbasestore.operation.handler.GetElementsHandler;
+import uk.gov.gchq.gaffer.hbasestore.operation.hdfs.handler.AddElementsFromHdfsHandler;
+import uk.gov.gchq.gaffer.hbasestore.retriever.HBaseRetriever;
+import uk.gov.gchq.gaffer.hbasestore.utils.TableUtils;
+import uk.gov.gchq.gaffer.hdfs.operation.AddElementsFromHdfs;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.Options;
+import uk.gov.gchq.gaffer.operation.graph.GraphFilters;
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.user.User;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Set;
+
+import static uk.gov.gchq.gaffer.store.StoreTrait.ORDERED;
+import static uk.gov.gchq.gaffer.store.StoreTrait.POST_AGGREGATION_FILTERING;
+import static uk.gov.gchq.gaffer.store.StoreTrait.POST_TRANSFORMATION_FILTERING;
+import static uk.gov.gchq.gaffer.store.StoreTrait.PRE_AGGREGATION_FILTERING;
+import static uk.gov.gchq.gaffer.store.StoreTrait.QUERY_AGGREGATION;
+import static uk.gov.gchq.gaffer.store.StoreTrait.STORE_AGGREGATION;
+import static uk.gov.gchq.gaffer.store.StoreTrait.STORE_VALIDATION;
+import static uk.gov.gchq.gaffer.store.StoreTrait.TRANSFORMATION;
+import static uk.gov.gchq.gaffer.store.StoreTrait.VISIBILITY;
+
+/**
+ * An HBase implementation of a Gaffer {@link Store}
+ * <p>
+ * The key detail of the HBase implementation is that any Edge inserted by a
+ * user is inserted into the hbase table twice, once with the source object
+ * being put first in the rowId and once with the destination bring put first in
+ * the rowId. This is to enable an edge to be found in a Range scan when providing
+ * only one end of the edge.
+ */
+public class HBaseStore extends Store {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HBaseStore.class);
+    public static final Set<StoreTrait> TRAITS =
```

Comment:
Standardise to match the AccumuloStore class (e.g. should create the Store traits Set in the same way etc)